ui/src/notebooks/index.ts

```diff
@@ -0,0 +1,41 @@
+import {FunctionComponent, ComponentClass} from 'react'
+
+export interface PipeProp {
+  idx: number
+}
+
+// NOTE: keep this interface as small as possible and
+// don't take extending it lightly. this should only
+// define what ALL pipe types require to be included
+// on the page.
+export interface TypeRegistration {
+  type: string // a unique string that identifies a pipe
```

Comment:
```suggestion
export interface TypeRegistration<T> {
  type: T // a unique string that identifies a pipe
```

Reply:
Would it be useful for the component registering this type to be able to define it as a `const` then the generic `string`?

Reply:
not here, but definitely in the serialization part where i currently have `type Pipe = any` with a todo to learn typescript. this part should have them all with the same shape to keep a consistent registration process, unless youre talking about some `as const` magic i'm unaware of